### PR TITLE
[docs] Add missing docstrings and fix doctests that arose as a result

### PR DESCRIPTION
**Pull request by odow**
Part of https://github.com/jump-dev/JuMP.jl/issues/1062

There are a couple of things that I will comment on in-line.

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -14,32 +19,59 @@ function link_example(content)
...
+    # don't necessarily want to document every MOI method.
+    #
+    # This is should be fine for now, because MOI doesn't export anything.
+    # However, also problematically, some doctests in MOI are not checked and
+    # are failing. Until they are fixed, we can't enable these options.
+    #
+    # TODO(odow): uncomment when possible.
+    # modules = [JuMP, MOI],
+    # checkdocs = :exports,
+    # ==========================================================================
```
Comment (odow) [inline docs/make.jl:72]: So we really want to enable these two options. But there are some failures in MOI doctests that I need to fix first.

File: docs/src/reference/variables.md [hunk 2]
```diff
@@ -49,4 +49,15 @@ num_variables
 VariableRef
 index(::VariableRef)
 optimizer_index(::VariableRef)
+
+add_variable
+check_belongs_to_model
+build_variable
+parse_one_operator_variable
```
Comment (odow) [inline docs/src/reference/variables.md:56]: This function is currently exported. Should it be?

File: docs/make.jl [hunk 3]
```diff
@@ -3,6 +3,11 @@ using Literate
 using JuMP
 using Test
 
+# Pass --fast as an argument to skip rebuilding the examples and running
+# doctests. Only use this argument to rapidly test small changes to the
+# Markdown. _Never_ set it in production.
+const _FAST = get(ARGS, 1, "") == "--fast"
```
Comment (odow) [inline docs/make.jl:9]: I can remove or split this into a separate PR if needed, but it was useful when compiling the list of missing docstrings, not to have to repeatedly generate the literate files.

Comment (blegat) [reply docs/make.jl:9]: Why not `findfirst(isequal("--fast"), ARGS) !== nothing` ?

File: docs/src/reference/variables.md (hunk 2 above)
Comment (blegat) [reply docs/src/reference/variables.md:56]: Yes: https://github.com/blegat/SetProg.jl/blob/64c056f8c22b3b79fdc0862c8938a0f23bd03796/src/macros.jl

File: docs/make.jl [hunk 4]
```diff
@@ -74,4 +106,5 @@ makedocs(
 
 deploydocs(
     repo   = "github.com/jump-dev/JuMP.jl.git",
+    push_preview = true,
```
Comment (blegat) [inline docs/make.jl:109]: I see that the doc of this PR is now available at https://jump.dev/JuMP.jl/previews/PR2393/, that's useful

**Review by mlubin — APPROVED** [review]

**Pull request merged**